Source/EasyNetQ.Management.Client/ManagementClient.cs

```diff
@@ -62,76 +57,47 @@ static ManagementClient()
         TimeSpan? timeout = null,
-        Action<HttpRequestMessage>? configureRequest = null,
+        Action<HttpRequestMessage>? configureHttpRequestMessage = null,
         bool ssl = false,
-        Action<HttpClientHandler>? handlerConfigurator = null
+        Action<HttpClientHandler>? configureHttpClientHandler = null
+    ) : this(
+        LegacyEndpointBuilder.Build(hostUrl, portNumber, ssl),
+        username,
+        password,
+        timeout,
+        configureHttpRequestMessage,
+        configureHttpClientHandler
     )
     {
-        if (string.IsNullOrEmpty(hostUrl))
-        {
-            throw new ArgumentException("hostUrl is null or empty");
-        }
-
-        if (hostUrl.StartsWith("https://"))
-            ssl = true;
-
-        if (ssl)
-        {
-            if (hostUrl.Contains("http://"))
-                throw new ArgumentException("hostUrl is illegal");
-            hostUrl = hostUrl.Contains("https://") ? hostUrl : "https://" + hostUrl;
-        }
-        else
-        {
-            if (hostUrl.Contains("https://"))
-                throw new ArgumentException("hostUrl is illegal");
-            hostUrl = hostUrl.Contains("http://") ? hostUrl : "http://" + hostUrl;
-        }
-
-        if (!urlRegex.IsMatch(hostUrl) || !Uri.TryCreate(hostUrl, UriKind.Absolute, out var urlUri))
-        {
-            throw new ArgumentException("hostUrl is illegal");
-        }
-
-        if (string.IsNullOrEmpty(username))
-        {
-            throw new ArgumentException("username is null or empty");
-        }
-
-        if (string.IsNullOrEmpty(password))
-        {
-            throw new ArgumentException("password is null or empty");
-        }
-
-        configureRequest ??= _ => { };
-
-        HostUrl = hostUrl;
-        Username = username;
-        PortNumber = portNumber;
-        this.configureRequest = configureRequest;
+    }
 
-        var httpHandler = new HttpClientHandler
-        {
-            Credentials = new NetworkCredential(username, password)
-        };
+    public ManagementClient(
+        Uri endpoint,
+        string username,
+        string password,
+        TimeSpan? timeout = null,
+        Action<HttpRequestMessage>? configureHttpRequestMessage = null,
+        Action<HttpClientHandler>? configureHttpClientHandler = null
+    )
+    {
+        if (!endpoint.IsAbsoluteUri) throw new ArgumentOutOfRangeException(nameof(endpoint), endpoint, "Endpoint should be absolute");
 
-        handlerConfigurator?.Invoke(httpHandler);
+        this.configureHttpRequestMessage = configureHttpRequestMessage;
 
-        httpClient = new HttpClient(httpHandler) { Timeout = timeout ?? defaultTimeout };
+        var httpHandler = new HttpClientHandler { Credentials = new NetworkCredential(username, password) };
+        configureHttpClientHandler?.Invoke(httpHandler);
+        httpClient = new HttpClient(httpHandler) { Timeout = timeout ?? DefaultTimeout, BaseAddress = endpoint };
```

Comment:
Consider to switch to `IHttpClientFactory` to get all rich configuration features on caller side. So you do not bother with `Action<HttpClientHandler>` when manually building `HttpClient`. Of course this will be one more dependency.

Reply:
I am fine with that if there is a simple way to wire up IHttpClientFactory in console app without DI.

Reply:
You may use some DI stuff inside like for `CreateBus` in ENQ. Consider something like that:
```
public class ManagementClient
 {
   public ManagementClient(...., Action<IHttpClientBuilder> configure)
   {
     var services = new ServiceCollection();
     var builder = services.AddHttpClient(...);
     configure(builder);
     provider = services.BuildServiceProvider(); // store into field, dispose field in Dispose() method
     httpClient = provider.GetRequiredService<IHttpClientFactory>().GetClient(nameof(ManagementClient));
    }
}
```

It another ctor accepting `Action<IHttpClientBuilder> configure` instead of `IHttpClientFactory`.